### PR TITLE
tools/meson: update to 0.59.2

### PR DESCRIPTION
**Pull request by neheb**
Update install procedure based on upstream feedback. Normally, meson is
to be installed with pip. But as pip is not mandated by the build
system, it cannot be used. Upstream provides a nice script to pack meson
automatically.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

ping @dhewg . No idea if these txt.in files should be placed elsewhere. Upstream also suggested not bothering to call python.

### CODE REVIEW

**Comment by dhewg** [comment]
Haven't tried this yet, but that looks nice, I don't think that script existed back then.
The location of the txt files doesn't really matter. lgtm, but if you think there's a more suitable place: go for it
And I'd keep the python binary in the cmdline, just to make sure it's using the same interpreter as the rest of the build system.

There's still that one leftover comment in meson.mk (HOST_BUILD_DEPENDS), mind removing that?

**Comment by neheb** [comment]
Removed HOST_BUILD_DEPENDS.

Made build files live in files instead of src/. There's no point in copying to HOST_BUILD_DIR.

**Comment by neheb** [comment]
I tested with CONFIG_ALL after wiping build/staging_dir. Everything compiled fine.

**Comment by eli-schwartz** [comment]
This script is a thin wrapper over python's builtin zipapp support. meson has supported zipapp deployment for a long time, but in https://github.com/mesonbuild/meson/commit/0fa808f7066d20047c0576da064bcc105053c401 I added a dedicated script which does this properly (i.e. "does not include the entire testsuite in the output").

If you add the `--interpreter` option to the packaging script, it will embed that as the shebang and it will be able to be run without explicitly invoking python first.

This was my original suggestion when I told @neheb about the zipapp support -- specifically, I suggested that you can use this option to embed the *correct* interpreter, i.e.

```
--interpreter "$(STAGING_DIR_HOST)/bin/$(PYTHON)"
```

This should work flawlessly unless those Make variables get changed to point elsewhere, but my assumption is you'd probably want to recreate your environment in that case.

Anyway, do whatever feels right to you. :)

**Comment by neheb** [comment]
Applied suggested change.

File: tools/meson/Makefile [hunk 1]
```diff
@@ -20,8 +20,11 @@ define Host/Compile
 endef
 
 define Host/Install
+	$(INSTALL_DIR) $(STAGING_DIR_HOST)/bin
+	$(HOST_BUILD_DIR)/packaging/create_zipapp.py $(HOST_BUILD_DIR) --interpreter $(STAGING_DIR_HOST)/bin/$(PYTHON) --outfile $(STAGING_DIR_HOST)/bin/meson
```
Comment (aparcar) [inline tools/meson/Makefile:24]: can you elaborate on this script? Do we happen to have that in tree or is this part of meson?

Comment (neheb) [reply tools/meson/Makefile:24]: it's part of meson. Hence the HOST_BUILD_DIR

**Comment by dhewg** [comment]
Answering myself:
Yeah, it's used for the shebang, but the vars get expanded

And I think that's fine

**Comment by neheb** [comment]
Of course

**Review by dhewg — APPROVED** [review]

**Comment by aparcar** [comment]
Thanks

**Pull request closed without merging**